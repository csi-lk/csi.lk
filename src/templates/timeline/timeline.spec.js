import React from 'react'

import TimelinePage from './timeline'

describe('Template: Home', () => {
  let props
  beforeEach(() => {
    props = {
      data: {
        content: {
          html: '<p>Test</p>',
          frontmatter: {
            description: 'Test',
            title: 'Test',
          },
        },
      },
      pathContext: {
        path: '/timeline',
      },
    }
  })
  it('renders', () => {
    const wrapper = shallow(<TimelinePage {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders preview', () => {
    const wrapper = shallow(<TimelinePage {...props} preview />)
    expect(wrapper).toMatchSnapshot()
  })
})
