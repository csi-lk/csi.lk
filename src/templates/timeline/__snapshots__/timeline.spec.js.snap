// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template: Home renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelinePage
    data={
      Object {
        "content": Object {
          "frontmatter": Object {
            "description": "Test",
            "title": "Test",
          },
          "html": "<p>Test</p>",
        },
      }
    }
    pathContext={
      Object {
        "path": "/timeline",
      }
    }
    preview={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SEOTagsComponent
          description="Test"
          path="/timeline"
          title="Test"
        />,
        <section
          className="timeline"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>Test</p>",
              }
            }
          />
        </section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Test",
          "path": "/timeline",
          "title": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>Test</p>",
              }
            }
          />,
          "className": "timeline",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<p>Test</p>",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SEOTagsComponent
            description="Test"
            path="/timeline"
            title="Test"
          />,
          <section
            className="timeline"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Test</p>",
                }
              }
            />
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Test",
            "path": "/timeline",
            "title": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Test</p>",
                }
              }
            />,
            "className": "timeline",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "dangerouslySetInnerHTML": Object {
                "__html": "<p>Test</p>",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Template: Home renders preview 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelinePage
    data={
      Object {
        "content": Object {
          "frontmatter": Object {
            "description": "Test",
            "title": "Test",
          },
          "html": "<p>Test</p>",
        },
      }
    }
    pathContext={
      Object {
        "path": "/timeline",
      }
    }
    preview={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SEOTagsComponent
          description="Test"
          path="/timeline"
          title="Test"
        />,
        <section
          className="timeline"
        >
          <div>
            &lt;p&gt;Test&lt;/p&gt;
          </div>
        </section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Test",
          "path": "/timeline",
          "title": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            &lt;p&gt;Test&lt;/p&gt;
          </div>,
          "className": "timeline",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<p>Test</p>",
          },
          "ref": null,
          "rendered": "<p>Test</p>",
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SEOTagsComponent
            description="Test"
            path="/timeline"
            title="Test"
          />,
          <section
            className="timeline"
          >
            <div>
              &lt;p&gt;Test&lt;/p&gt;
            </div>
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Test",
            "path": "/timeline",
            "title": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              &lt;p&gt;Test&lt;/p&gt;
            </div>,
            "className": "timeline",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "<p>Test</p>",
            },
            "ref": null,
            "rendered": "<p>Test</p>",
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
